File: 2020/1.2.rb
require_relative '1.1.rb'

module DayOnePartTwo
  module_function

  def solve(file)
    inputs = DayOnePartOne.prepare_inputs(file)
    find_2020_sum(inputs)
  end

  def find_2020_sum(inputs)
    inputs.each do |a|
      inputs.each do |b|
        inputs.each do |c|
          return a * b * c if a + b + c == 2020
        end
      end
    end
  end
end

# puts DayOnePartTwo.solve(__dir__ + '/1test.txt')
puts DayOnePartTwo.solve(__dir__ + '/1.txt')

File: 2021/01/1.2.php
<?php


require "1.1.php";

class DayOnePartTwo
{
    function solve($file)
    {
        $inputs = DayOnePartOne::prepareInputs($file);
        echo $this->countIncreasingWindows($inputs, 3) . "\n";
    }

    function countIncreasingWindows($inputs, $size)
    {
        $previous = null;
        $total = 0;
        foreach ($inputs as $key => $input) {
            if ($key) {
                $window = array_slice($inputs, $key,  $size);
                $input = array_sum($window);
                $total += $previous && $input > $previous ? 1 : 0;
                $previous = $input;
            }
        }
        return $total;
    }
}

$class = new DayOnePartTwo();
// $class->solve("/2021/1test.txt");
$class->solve("/2021/1.txt");
